pkg/pool/pool.go

```diff
@@ -65,16 +65,8 @@ func NewWithContext(ctx context.Context, workers, capacity int) (Interface, cont
 	// 3. marks work as done in our sync.WaitGroup.
 	for idx := 0; idx < workers; idx++ {
 		go func() {
-			for {
-				select {
-				case <-ctx.Done():
```

Comment:
We need to keep `<-ctx.Done()` I think.
I'd just replace `break` with `return`.

Reply:
I had that before. It stops processing already queued work immediately though. All the error tests are actually wrong in that case.

We could drop all work if ctx is already done but it feels wrong because the Go call can‘t surface that your work will never be worked on. In that sense I‘d say already queued work should pass happily.